pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -260,6 +260,26 @@ func (f *ring1Factory) LogsForObject(object, options runtime.Object) (*restclien
 
 		return clientset.Core().Pods(pod.Namespace).GetLogs(pod.Name, opts), nil
 
+	case *extensions.Deployment:
+		opts, ok := options.(*api.PodLogOptions)
+		if !ok {
+			return nil, errors.New("provided options object is not a PodLogOptions")
+		}
+		selector, err := metav1.LabelSelectorAsSelector(t.Spec.Selector)
+		if err != nil {
+			return nil, fmt.Errorf("invalid label selector: %v", err)
+		}
+		sortBy := func(pods []*v1.Pod) sort.Interface { return controller.ByLogging(pods) }
+		pod, numPods, err := GetFirstPod(clientset.Core(), t.Namespace, selector, 20*time.Second, sortBy)
```

Comment:
Yes, consistent with the others.  See https://github.com/kubernetes/kubernetes/issues/19873 for proposed extension.